packages/@aws-cdk/aws-batch/README.md

```diff
@@ -13,39 +14,271 @@
+
+```ts
+const myLaunchTemplate = new ec2.CfnLaunchTemplate(this, 'LaunchTemplate', {
+  managed: true,
+  computeResources: {
+    type: batch.ComputeResourceType.SPOT,
+    // Bid at 75% of the on-demand price
+    bidPercentage: 75,
+    // Limits the minimum/maximum size of our spotfleet will be
+    maxvCpus: 4,
+    minvCpus: 1,
+    vpc,
+  },
+});
+```
+
+### Importing an existing Compute Environment
+
+To import an existing batch compute environment, call `ComputeEnvironment.fromComputeEnvironmentArn()`.
+
+Below is an example:
+
+```ts
+const computeEnv = ComputeEnvironment.fromComputeEnvironmentArn(this, 'imported-compute-env', 'arn:aws:batch:us-east-1:555555555555:compute-environment/My-Compute-Env');
+```
+
+## Job Queue
+
+Batch Job Queues store incoming requests before assignment to an available compute environment.
+
+```ts
+const jobQueue = new batch.JobQueue(stack, 'JobQueue', {
+  computeEnvironments: [
+    {
+      // Defines a collection of compute resources to handle assigned batch jobs
+      computeEnvironment,
+      // Order determines the allocation order for jobs (i.e. Lower means higher preferance for job assignment)
+      order: 1,
+    },
+  ],
+});
+```
+
+### Priorty-Based Queue Example
+
+Sometimes you might come across a situation where certain batch jobs need to be processed before others. In this case, creating priority based queues can achieve this. Below is an example of how that can be done:
+
+```ts
+const highPrioQueue = new batch.JobQueue(stack, 'JobQueue', {
+  computeEnvironments: sharedComputeEnvs,
+  priority: 2,
+});
+
+const lowPrioQueue = new batch.JobQueue(stack, 'JobQueue', {
+  computeEnvironments: sharedComputeEnvs,
+  priority: 1,
+});
+```
+
+In this example, both job queues share the same set of compute environments but our `highPrioQueue` will have its jobs assigned first. Even though our other queue might have jobs waiting to be assigned, they will be ignored until our `highPrioQueue` has no jobs waiting for assignment.
+
+### Importing an existing Job Queue
+
+To import an existing batch job queue, call `JobQueue.fromJobQueueArn()`.
+
+Below is an example:
+
+```ts
+const jobQueue = JobQueue.fromJobQueueArn(this, 'imported-job-queue', 'arn:aws:batch:us-east-1:555555555555:job-queue/High-Prio-Queue');
+```
+
+## Job Definition
+
+A Batch Job definition helps AWS Batch understand important details about how to run your application in the scope of a Batch Job. This involves key information like resource requirements, what containers to run, how the compute environment should be prepared, and more. Below is a simple example of how to create a job definition:
+
+```ts
+const repo = new ecr.Repository(stack, 'batch-job-repo');
```

Comment:
Maybe use an existing repo as an example? It more clearly shows a link to a specific image:

```suggestion
const repo = ecr.Repository.fromRepositoryName(stack, 'batch-job-repo', 'todo-list');
```